src/main/java/com/squareup/javapoet/TypeSpec.java

```diff
@@ -234,6 +242,10 @@ void emit(CodeWriter codeWriter, String enumName, Set<Modifier> implicitModifier
     }
   }
 
+  public boolean isAnnotationDeclaration() {
```

Comment:
move this to be under `hasModifier` ? Might as well put these methods all together.